src/core/js/models/itemsComponentModel.js

```diff
@@ -0,0 +1,76 @@
+define([
+    'core/js/models/componentModel',
+    'core/js/models/itemModel'
+], function(ComponentModel, ItemModel) {
+
+    var ItemsComponentModel = ComponentModel.extend({
+
+        toJSON: function() {
+            var json = _.clone(this.attributes);
+
+            json._items = this.get('_items').toJSON();
+
+            return json;
+        },
+
+        init: function() {
+            this.setUpItems();
+
+            this.listenTo(this.get('_items'), {
+                'change:_isVisited': this.checkCompletionStatus
+            });
+        },
+
+        setUpItems: function() {
+            var items = this.get('_items');
+            var collection = new Backbone.Collection(null, { model: ItemModel });
+
+            items.forEach(function(item, index) {
```

Comment:
This doesn't make use of the array returned by calling .map so .forEach could still be used. If immutability is desirable though
var items = items.slice().map(
would prevent mutation

Also since there's now only one .add call why not just use

var collection = new Backbone.Collection(items, { model: itemModel })

After the array has been mapped.

Sorry for the lack of MD I'm on a phone currently.